test/iio-oscilloscope/test_adi_oscilloscope.py

```diff
@@ -0,0 +1,46 @@
+import pytest
+import pyguit
+import time
+import os
+import shutil
+
+class TestIIOOscilloscope:
+
+    @classmethod
+    def setup_class(self):
+        '''Setup the pyguit object. Will be called once before the start of test'''
+        self.gui = pyguit.gui()
+        self.gui.attach_openbox()
+        # self.gui.run_ssh_agent()
+        time.sleep(10)
+        try:
+            os.makedirs("results")
+        except FileExistsError:
+            # directory already exists
+            shutil.rmtree('results')
+            os.makedirs("results")
+        
+    @classmethod
+    def teardown_class(self):
+        '''Garbage collector. Will be called once after running of tests'''
+        time.sleep(10)
+        self.gui.dettach_openbox()
+        del self.gui
+
+    @pytest.mark.remote
+    def test_open_app_on_remote(self, ip, delay):
+        '''Test if app opens, and checks main window'''
+        self.gui.open_app(
+            host=ip,
+            user="analog",
+            app_name="osc",
+            path="/usr/local/bin/osc",
+        )
+        time.sleep(delay)
+        #find_active_screen
+        for w in self.gui.get_open_windows():
+            if w:   
+                print(self.gui.get_window_title(w))
+        time.sleep(delay)
+        self.gui.controller.screenshot("results/test_open_a_app.png")
+      
```

Comment:
Is the scope of the test until screenshot only? I we could atleast compare 1 image to add extra check would be helpfuk

Reply:
this is now okay.

Reply:
This is expected to only take screenshot and no assertions?

Reply:
This function can be found in section 'Perform actions based on the found window, line 63.
 assert self.gui.controller.locateOnScreen("ref_test_open_app.png", grayscale=True, confidence=0.9)

Reply:
I am assuming this test file is a WIP and will be completed in another PR,